src/main/java/uk/gov/mca/beacons/service/registrations/RegistrationsService.java

```diff
@@ -1,27 +1,98 @@
 package uk.gov.mca.beacons.service.registrations;
 
+import java.util.List;
+import java.util.UUID;
+import javax.transaction.Transactional;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+import uk.gov.mca.beacons.service.model.Beacon;
+import uk.gov.mca.beacons.service.model.BeaconPerson;
+import uk.gov.mca.beacons.service.model.BeaconStatus;
+import uk.gov.mca.beacons.service.model.BeaconUse;
+import uk.gov.mca.beacons.service.model.PersonType;
 import uk.gov.mca.beacons.service.model.Registration;
+import uk.gov.mca.beacons.service.repository.BeaconPersonRepository;
+import uk.gov.mca.beacons.service.repository.BeaconRepository;
+import uk.gov.mca.beacons.service.repository.BeaconUseRepository;
 
 @Service
+@Transactional
 public class RegistrationsService {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(
     RegistrationsService.class
   );
 
-  public Registration register(Registration registration) {
-    // TODO: Persist beacon, owner etc. to database, send confirmation email
+  private final BeaconRepository beaconRepository;
+  private final BeaconUseRepository beaconUseRepository;
+  private final BeaconPersonRepository beaconPersonRepository;
+
+  @Autowired
+  public RegistrationsService(
+    BeaconRepository beaconRepository,
+    BeaconUseRepository beaconUseRepository,
+    BeaconPersonRepository beaconPersonRepository
+  ) {
+    this.beaconRepository = beaconRepository;
+    this.beaconPersonRepository = beaconPersonRepository;
+    this.beaconUseRepository = beaconUseRepository;
+  }
 
+  public Registration register(Registration registration) {
     LOGGER.info("Attempting to persist registration {}", registration);
 
-    // Create new BeaconPerson to represent owner, persist
-    // For each beacon posted, create new Beacon and persist
-    // -----> For each use associated with Beacon, create new BeaconUse and persist
-    // For each beacon posted, create new Beacon and persist
+    final List<Beacon> beacons = registration.getBeacons();
+    beacons.forEach(this::registerBeacon);
 
     return registration;
   }
+
+  private void registerBeacon(Beacon beacon) {
+    beacon.setBeaconStatus(BeaconStatus.NEW);
+    final Beacon persistedBeacon = beaconRepository.save(beacon);
+
+    registerBeaconUses(persistedBeacon);
+    registerOwner(persistedBeacon);
+    registerEmergencyContacts(persistedBeacon);
+  }
+
+  private void registerBeaconUses(Beacon beacon) {
+    final List<BeaconUse> beaconUses = beacon.getUses();
+    final UUID beaconId = beacon.getId();
+
+    beaconUses.forEach(
```

Comment:
I was in two minds here as to how we set the `mainUse` on a beacon. Options were:

- Set the first use as the `mainUse` - this felt a little restrictive if the consumer of the API wanted to specify which beacon use was the main use
- Allow the consumer to specify the `mainUse`

I went with the latter, but it won't be too much work to change the test case to assert that the first use is always the main.

@stuart-madetech @zackads @NyWeb thoughts?

Reply:
I agree with choice of allowing consumer to specify as it's more explicit.  Using the index position in the `Use` array to signify `mainUse` would rely on documentation for the consumer to understand the significance, could be flaky?